test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js

```diff
@@ -1070,8 +1070,91 @@ describe( `[${ host }] Calypso Gutenberg Site Editor Tracking: (${ screenSize })
 				0,
 				"detaching blocks from template part shouldn't trigger replace blocks event"
 			);
+		} );
+
+		describe( 'tracks "entity_context" property for block events in a template part', function () {
+			it( 'For "wpcom_block_inserted" event', async function () {
+				// Reload editor to start from consistent clean slate for tests.
+				await this.driver.navigate().refresh();
+				await driverHelper.acceptAlertIfPresent( this.driver );
+
+				const editor = await SiteEditorComponent.Expect( this.driver );
+				await editor.waitForTemplateToLoad();
+				await editor.waitForTemplatePartsToLoad();
+				await clearEventsStack( this.driver );
+
+				// Insert a block at the end of the template part.
+				await editor.runInCanvas( async () => {
+					await driverHelper.clickWhenClickable( this.driver, By.css( '.wp-block-template-part' ) );
+					await driverHelper.clickWhenClickable(
+						this.driver,
+						By.css( '.block-editor-inserter__toggle' )
+					);
+				} );
+				const quickInserterSearchInputLocator = By.css(
+					'.block-editor-inserter__quick-inserter .block-editor-inserter__search-input'
+				);
+				const blockItemLocator = By.css(
+					'.block-editor-inserter__quick-inserter .block-editor-block-types-list__item'
+				);
+				await driverHelper.setWhenSettable(
+					this.driver,
+					quickInserterSearchInputLocator,
+					'Heading'
+				);
+				await driverHelper.clickWhenClickable( this.driver, blockItemLocator );
+
+				const insertedEvents = ( await getEventsStack( this.driver ) ).filter(
+					( event ) => event[ 0 ] === 'wpcom_block_inserted'
+				);
+
+				assert.strictEqual( insertedEvents.length, 1 );
+				assert.strictEqual( insertedEvents[ 0 ][ 1 ].entity_context, 'core/template-part/header' );
+				assert.strictEqual( insertedEvents[ 0 ][ 1 ].template_part_id, 'pub/tt1-blocks//header' );
+			} );
+
+			it( 'For "wpcom_block_moved_*" events', async function () {
+				await SiteEditorComponent.Expect( this.driver );
+
+				await driverHelper.clickWhenClickable(
+					this.driver,
+					By.css( '.block-editor-block-mover-button.is-up-button' )
+				);
+				await driverHelper.clickWhenClickable(
+					this.driver,
+					By.css( '.block-editor-block-mover-button.is-down-button' )
+				);
+
+				const events = await getEventsStack( this.driver );
+				assert.strictEqual( events.length, 2 );
 
-			await deleteCustomEntities( this.driver, 'wp_template_part' );
```

Comment:
We can get rid of this from where it was.  Its not necessary to clean the template parts at that step in order for that test to pass again if ran.  Therefore, we can save this deletion until the normal cleanup stage to reduce its chance of conflicting with another parallel test run on this site.